src/node_contextify.cc

```diff
@@ -507,6 +510,208 @@ class ContextifyContext {
 
     args.GetReturnValue().Set(ctx->sandbox()->GetPropertyNames());
   }
+
+  static void IndexedPropertyGetterCallback(
+      uint32_t index,
+      const PropertyCallbackInfo<Value>& args) {
+    ContextifyContext* ctx;
+    ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Data().As<Object>());
+
+    // Still initializing
+    if (ctx->context_.IsEmpty())
+      return;
+
+    Local<Context> context = ctx->context();
+    v8::Isolate* isolate = context->GetIsolate();
+    Local<Object> sandbox = ctx->sandbox();
+    MaybeLocal<Value> maybe_rv =
+        sandbox
+            ->GetRealNamedProperty(context, UIntToName(isolate, index));
+    if (maybe_rv.IsEmpty()) {
+        maybe_rv = ctx->global_proxy()
+            ->GetRealNamedProperty(context, UIntToName(isolate, index));
+    }
+
+    Local<Value> rv;
+    if (maybe_rv.ToLocal(&rv)) {
+      if (rv == sandbox)
+        rv = ctx->global_proxy();
+
+      args.GetReturnValue().Set(rv);
+    }
+  }
+
+
+  static void IndexedPropertySetterCallback(
+      uint32_t index,
+      Local<Value> value,
+      const PropertyCallbackInfo<Value>& args) {
+    ContextifyContext* ctx;
+    ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Data().As<Object>());
+
+    // Still initializing
+    if (ctx->context_.IsEmpty())
+      return;
+
+    Local<Context> context = ctx->context();
+    v8::Isolate* isolate = context->GetIsolate();
+
+    auto attributes = PropertyAttribute::None;
+    bool is_declared =
+        ctx->global_proxy()
+            ->GetRealNamedPropertyAttributes(ctx->context(),
+                 UIntToName(isolate, index))
+        .To(&attributes);
+    bool read_only =
+        static_cast<int>(attributes) &
+        static_cast<int>(PropertyAttribute::ReadOnly);
+
+    if (is_declared && read_only)
+      return;
+
+    // true for x = 5
+    // false for this.x = 5
+    // false for Object.defineProperty(this, 'foo', ...)
+    // false for vmResult.x = 5 where vmResult = vm.runInContext();
+    bool is_contextual_store = ctx->global_proxy() != args.This();
+
+    // Indicator to not return before setting (undeclared) function declarations
+    // on the sandbox in strict mode, i.e. args.ShouldThrowOnError() = true.
+    // True for 'function f() {}', 'this.f = function() {}',
+    // 'var f = function()'.
+    // In effect only for 'function f() {}' because
+    // var f = function(), is_declared = true
+    // this.f = function() {}, is_contextual_store = false.
+    bool is_function = value->IsFunction();
+
+    if (!is_declared && args.ShouldThrowOnError() && is_contextual_store &&
+    !is_function)
```

Comment:
You fixed the style for this further up but not here.

Reply:
Done.